### PR TITLE
tools: add fpgainfo for Darby Creek & Rush Creek

### PR DESCRIPTION
**Pull request by anandaravuri**

### CODE REVIEW

File: tools/libboard/board_dc/board_dc.c [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+	NIOS II Firmware Build 0x0 32 RW[23:0] 24 hFFFFFF Build version of NIOS II Firmware
+	NIOS FW is up e.g. 1.0.1 for first release
+	[31:24] 8hFF Firmware Support Revision - ASCII code
+	0xFF is the default value without NIOS FW, will be changed after NIOS FW is up
+	*/
+
+	errno = 0;
+	var = strtoul(buf, &endptr, 16);
+	if (var == 0  &&
+		errno != 0) {
```
Comment (tswhison) [inline tools/libboard/board_dc/board_dc.c]: I think the validity check for strtoul( ) conversion is to make sure endptr has been set appropriately.
You can do
```c
if (endptr && *endptr == '\0') {
```
or
```c
if (endptr = buf + strlen(buf)) {
```

File: tools/libboard/board_dc/board_dc.c [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+	fpga_result res                      = FPGA_OK;
+	char bmc_ver[FPGA_VAR_BUF_LEN]       = { 0 };
+	char max10_ver[FPGA_VAR_BUF_LEN]     = { 0 };
+	char mac_str[18] = { 0 };
+	struct ether_addr MAC ;
+
+
+	res = read_bmcfw_version(token, bmc_ver, FPGA_VAR_BUF_LEN);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read bmc version");
```
Comment (tswhison) [inline tools/libboard/board_dc/board_dc.c]: Return res?

File: tools/libboard/board_dc/board_dc.c [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+	}
+
+	res = read_max10fw_version(token, max10_ver, FPGA_VAR_BUF_LEN);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read max10 version");
```
Comment (tswhison) [inline tools/libboard/board_dc/board_dc.c]: Return res?

File: tools/libboard/board_dc/board_dc.c [hunk 4]
```diff
@@ -0,0 +1,326 @@
+// Copyright(c) 2019, Intel Corporation
```
Comment (tswhison) [inline tools/libboard/board_dc/board_dc.c]: 2019-2020

File: tools/libboard/board_dc/board_dc.h [hunk 5]
```diff
@@ -0,0 +1,104 @@
+// Copyright(c) 2019, Intel Corporation
```
Comment (tswhison) [inline tools/libboard/board_dc/board_dc.h]: 2019-2020

File: tools/libboard/board_rc/board_rc.c [hunk 6]
```diff
@@ -333,5 +346,125 @@ fpga_result print_board_info(fpga_token token)
...
+
+	res = fpgaObjectGetSize(sec_object, &size, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get object size ");
+		resval = res;
+		goto out_destroy;
+	}
+
+	if (size > FPGA_STR_SIZE) {
+		FPGA_ERR("object size bigger then buffer size");
```
Comment (tswhison) [inline tools/libboard/board_rc/board_rc.c]: s,FPGA_ERR,OPAE_ERR,

File: tools/libboard/board_rc/board_rc.c [hunk 7]
```diff
@@ -333,5 +346,125 @@ fpga_result print_board_info(fpga_token token)
...
+	if (read_sysfs(token, SYSFS_TCM_SR_CANCEL, name) == FPGA_OK)
+		printf("FIM CSK IDs canceled : %s", strlen(name) > 1 ? name : "None\n");
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_BMC_CANCEL, name) == FPGA_OK)
+		printf("BMC CSK IDs canceled: %s", strlen(name) > 1 ? name : "None\n");
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_PR_CANCEL, name) == FPGA_OK)
+		printf("AFU CSK IDs canceled: %s", strlen(name) > 1 ? name : "None\n");
```
Comment (tswhison) [inline tools/libboard/board_rc/board_rc.c]: Silent errors when read_sysfs doesn't return FPGA_OK?

File: tools/libboard/board_dc/board_dc.c [hunk 8]
```diff
@@ -130,8 +130,7 @@ fpga_result parse_fw_ver(char *buf, char *fw_ver, size_t len)
 
 	errno = 0;
 	var = strtoul(buf, &endptr, 16);
-	if (var == 0  &&
-		errno != 0) {
+	if (endptr && *endptr == '\0') {
```
Comment (tswhison) [inline tools/libboard/board_dc/board_dc.c]: Whoops, this is the success case, so you would need:
```c
if (!endptr || *endptr != '\0') {
```

File: tools/libboard/board_dc/board_dc.c (hunk 3 above)
Comment (anandaravuri) [reply tools/libboard/board_dc/board_dc.c]: fixed

File: tools/libboard/board_dc/board_dc.c (hunk 4 above)
Comment (anandaravuri) [reply tools/libboard/board_dc/board_dc.c]: fixed

File: tools/libboard/board_dc/board_dc.h (hunk 5 above)
Comment (anandaravuri) [reply tools/libboard/board_dc/board_dc.h]: fixed

File: tools/libboard/board_rc/board_rc.c (hunk 6 above)
Comment (anandaravuri) [reply tools/libboard/board_rc/board_rc.c]: fixed

File: tools/libboard/board_rc/board_rc.c (hunk 7 above)
Comment (anandaravuri) [reply tools/libboard/board_rc/board_rc.c]: fixed

File: tools/libboard/board_dc/board_dc.c (hunk 8 above)
Comment (anandaravuri) [reply tools/libboard/board_dc/board_dc.c]: fixed

File: tools/libboard/board_dc/board_dc.c [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+	/* BMC FW version format reading
+	NIOS II Firmware Build 0x0 32 RW[23:0] 24 hFFFFFF Build version of NIOS II Firmware
+	NIOS FW is up e.g. 1.0.1 for first release
+	[31:24] 8hFF Firmware Support Revision - ASCII code
+	0xFF is the default value without NIOS FW, will be changed after NIOS FW is up
+	*/
+
+	errno = 0;
+	var = strtoul(buf, &endptr, 16);
+	if (!endptr || *endptr == '\0') {
```
Comment (tswhison) [inline tools/libboard/board_dc/board_dc.c]: The check should be
```c
if (!endptr || *endptr != '\0') {
```

Comment (tswhison) [reply tools/libboard/board_dc/board_dc.c]: In other words, on success, endptr will be set to the null terminator of the input string.  If it fails, then endptr points to the character that failed to parse.

**Pull request merged**